backend/functions/tasks.py

```diff
@@ -88,12 +88,12 @@ def sentence_text_translate_and_save_translation_pairs(
     # Iterate through the languages
     for output_language in languages:
 
-        # Create a TranslationPair object list
-        translation_pair_objects = []
-
         # Loop through all the sentences to be translated in batch format
         for i in range(0, len(all_sentences_to_be_translated), batch_size):
 
+            # Create a TranslationPair object list
+            translation_pair_objects = []
+

```

Comment:
Why is this required?

Wasn't the solution just de-indenting line number 205?

Reply:
Sorry, did not notice your PR comment.

> due to overhead of uploading so I am not sure 

What overhead?

Reply:
OK, so you are saying, if we are to bulk_create say 1 lakh objects, it might cause some issues.

Yeah, this depends on the memory usage. 
For example, assuming each record is 1kB (ideally it wouldn't be that much) and 1L records, it would take 100MB.
If 1million records, 1GB of RAM.

Reply:
Yeah, both of them are valid ways. I don't know if it will exactly lead to an issue so that's what I wanted to discuss if it will not lead to an issue then I can just de-indent line 205.

Reply:
However, doing it in batch-wise uploads might be better to avoid system memory usage. Since this is our low-priority queue, we would not want to spend a lot of memory on upload alone.

Reply:
Makes sense. But performing bulk_create just for a batch (of 128 records) isn't going to be very much useful than just iterating and saving. So wondering what we can do here.

Either, we set a limit on maximum records per upload. Or just continue doing this.
I am fine with both.

Reply:
> For example, assuming each record is 1kB (ideally it wouldn't be that much) and 1L records, it would take 100MB.
If 1million records, 1GB of RAM.

Actually, this happened in one of the earlier cases.
Please check the results of this task: 
`
Traceback (most recent call last):
File "/usr/local/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
return self.cursor.execute(sql, params)
psycopg2.errors.OutOfMemory: out of memory
DETAIL: Failed on request of size 1048576 in memory context "ErrorContext".


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
File "/usr/local/lib/python3.7/site-packages/celery/app/trace.py", line 451, in trace_task
R = retval = fun(*args, **kwargs)
File "/usr/local/lib/python3.7/site-packages/celery/app/trace.py", line 734, in __protected_call__
return self.run(*args, **kwargs)
File "/usr/src/backend/functions/tasks.py", line 204, in sentence_text_translate_and_save_translation_pairs
multi_inheritance_table_bulk_insert(translation_pair_objects)
File "/usr/src/backend/utils/custom_bulk_create.py", line 28, in multi_inheritance_table_bulk_insert
parent_model.objects.bulk_create(parent_objects)
File "/usr/local/lib/python3.7/site-packages/django/db/models/manager.py", line 85, in manager_method
return getattr(self.get_queryset(), name)(*args, **kwargs)
File "/usr/local/lib/python3.7/site-packages/django/db/models/query.py", line 507, in bulk_create
objs_without_pk, fields, batch_size, ignore_conflicts=ignore_conflicts,
File "/usr/local/lib/python3.7/site-packages/django/db/models/query.py", line 1274, in _batched_insert
ignore_conflicts=ignore_conflicts,
File "/usr/local/lib/python3.7/site-packages/django/db/models/query.py", line 1254, in _insert
return query.get_compiler(using=using).execute_sql(returning_fields)
File "/usr/local/lib/python3.7/site-packages/django/db/models/sql/compiler.py", line 1397, in execute_sql
cursor.execute(sql, params)
File "/usr/local/lib/python3.7/site-packages/django/db/backends/utils.py", line 66, in execute
return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
File "/usr/local/lib/python3.7/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
return executor(sql, params, many, context)
File "/usr/local/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
return self.cursor.execute(sql, params)
File "/usr/local/lib/python3.7/site-packages/django/db/utils.py", line 90, in __exit__
raise dj_exc_value.with_traceback(traceback) from exc_value
File "/usr/local/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
return self.cursor.execute(sql, params)
django.db.utils.OperationalError: out of memory
DETAIL: Failed on request of size 1048576 in memory context "ErrorContext".
`

Reply:
> Failed on request of size 1048576 in memory

It says failed for 1MB. Not sure why it can't even handle that.

Reply:
These were ~330000 sentences..
Took 8 hours for the job to run on indicTrans. Not sure if that's 1 MB